content/components/button-group.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Button group
+description: Button group renders a series of buttons.
+reactId: button_group
+figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=20363%3A71038&t=GrA68FvIJRoUaVbl-1
+rails: https://primer.style/view-components/components/beta/buttongroup
+---
+
+import {Box, Text} from '@primer/react'
+import ComponentLayout from '~/src/layouts/component-layout'
+export default ComponentLayout
+
+<img
+    src="https://user-images.githubusercontent.com/586552/226987730-84d99d46-1dd4-4ddc-91b3-046d7eeedbad.png"
+    role="presentation"
+    width="960"
+    alt="An image showing a button group in medium and small sizes."
+/>
+
+## Usage
+
+Buttons can be grouped together as individual segments of related actions. 
+
+Each segment in a button group is comprised from our default button component and can be visually represented with the same button types and states.
+
+Grouping buttons with a button group is better than rendering buttons close together for the following uses:
+
+- grouping a set of closely related buttons with other less related buttons
+- grouping multiple sets of buttons
+- saving horizontal space when rendering multiple closely-related buttons
+
+### Best practices
+
+- Use button groups to organize similar functionality. Don't group buttons just because they're close together
+
+- For most use-cases, only default button types should be used in button groups. In rare cases, primary buttons can be included in button groups but there should only ever be one primary button (if any) in a button group.
```

Comment:
Non-blocking nitpick: I think we need to remove the returns between each bullet point for these to be rendered as `li`s in a a single `ul`. We can check the preview deployment in a few months when it finishes 🙃